programs/drift/src/controller/amm.rs

```diff
@@ -382,6 +382,112 @@ pub fn get_fee_pool_tokens(
     )?
     .cast()
 }
+fn calculate_revenue_pool_transfer(
+    market: &PerpMarket,
+    spot_market: &SpotMarket,
+    amm_fee_pool_token_amount_after: u128,
+    terminal_state_surplus: i128,
+) -> DriftResult<i128> {
+    validate!(market
```

Comment:
should this be updated when the `revenue_withdraw_since_last_settle` is actually updated? feel like this won't stop the tx that corrupts the state, will just brick it afterward

Reply:
moved into `validate_perp_market`